src/System.Runtime/tests/System/Buffer.cs

```diff
@@ -4,111 +4,211 @@
 
 using System;
 using System.Collections.Generic;
-using System.Reflection;
-
 using Xunit;
 
-public static unsafe class BufferTests
+public static class BufferTests
 {
+    public static IEnumerable<object[]> BlockCopy_TestData()
+    {
+        yield return new object[] { new byte[] { 0x1a, 0x2b, 0x3c, 0x4d }, 0, new byte[] { 0x6f, 0x6f, 0x6f, 0x6f, 0x6f, 0x6f }, 1, 4, new byte[] { 0x6f, 0x1a, 0x2b, 0x3c, 0x4d, 0x6f } };
+
+        var dst = new byte[] { 0x1a, 0x2b, 0x3c, 0x4d, 0x5e };
+        yield return new object[] { dst, 1, dst, 2, 2, new byte[] { 0x1a, 0x2b, 0x2b, 0x3c, 0x5e } };
```

Comment:
These are long lines with multiple nested arrays.  At the very least, I'd suggest splitting across multiple lines with comments to indicate what each piece is, e.g. instead of:
```
yield return new object[] { new byte[] { 0x1a, 0x2b, 0x3c, 0x4d }, 0, new byte[] { 0x6f, 0x6f, 0x6f, 0x6f, 0x6f, 0x6f }, 1, 4, new byte[] { 0x6f, 0x1a, 0x2b, 0x3c, 0x4d, 0x6f } };
```
do:
```
yield return new object[] { 
    new byte[] { 0x1a, 0x2b, 0x3c, 0x4d }, // src
    0, // srcOffset
    new byte[] { 0x6f, 0x6f, 0x6f, 0x6f, 0x6f, 0x6f }, // dst
    1, // dstOffset
    4, // count
    new byte[] { 0x6f, 0x1a, 0x2b, 0x3c, 0x4d, 0x6f } // expected
};
```
But, given that there are only two inputs here, it might be better to just have a single [Fact] that contains these... there's more ceremony around the enumerable, yield return, etc. than there is in just duplicating the Buffer.BlockCopy and Assert.Equal calls.